Updated redirects

sachinSingh16-09 opened the pull request:
#### References to other Issues or PRs
#24140 
#### Brief description of what is fixed or changed
Updated a bunch of permanent redirects

#### Other comments


#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

sachinSingh16-09 committed:
Updated redirects

haru-44 commented:
I think the DOI link should not be modified.

https://github.com/sympy/sympy/pull/25505#discussion_r1293051942

sachinSingh16-09 commented:
Ohkk I will make the changes..

sachinSingh16-09 committed:
Update philosophy.rst

sachinSingh16-09 committed:
Update quaternion.py

sachinSingh16-09 committed:
Update iterables.py

sachinSingh16-09 committed:
Update group_numbers.py

sachinSingh16-09 committed:
Update group_numbers.py

sachinSingh16-09 committed:
Update group_numbers.py

sachinSingh16-09 committed:
Update primetest.py

sachinSingh16-09 committed:
Update primetest.py

sachinSingh16-09 committed:
Update primetest.py

sachinSingh16-09 committed:
Update primetest.py

sachinSingh16-09 committed:
Update primetest.py

sachinSingh16-09 committed:
Update diophantine.py

sachinSingh16-09 committed:
Update diophantine.py

sachinSingh16-09 committed:
Update diophantine.py

sachinSingh16-09 committed:
Update diophantine.py

haru-44 commented:
* "https://help.github.com/en/articles/set-up-git" redirects to "https://docs.github.com/en/get-started/getting-started-with-git/set-up-git"
* Links to "docs.github.com" should use "https://docs.github.com/foobar" instead of "https://docs.github.com/en/foobar". Redirect to a page appropriate for the user's language environment.
* "https://mae.ufl.edu/~fregly/PDFs/autolev_tutorial.pdf" is probably a broken link.
* Why did you fix "http://www.eecis.udel.edu/~saunders/papers/sffge/it5.ps" ?

sachinSingh16-09 committed:
Update matrices.py

sachinSingh16-09 committed:
Update dev-setup.md

sachinSingh16-09 committed:
Update workflow-process.md

sachinSingh16-09 committed:
Update workflow-process.md

sachinSingh16-09 commented:
- "https://help.github.com/en/articles/set-up-git" redirects to "https://docs.github.com/en/get-started/getting-started-with-git/set-up-git" - it's already changed
<img width="601" alt="Screenshot 2024-01-19 193650" src="https://github.com/sympy/sympy/assets/97588030/c0a89144-3533-4b0f-82bf-c966ae2f96ae">

- Links to "docs.github.com" should use "https://docs.github.com/foobar" instead of "https://docs.github.com/en/foobar". Redirect to a page appropriate for the user's language environment. - I have made the changes, Thanks.

- "https://mae.ufl.edu/~fregly/PDFs/autolev_tutorial.pdf" is probably a broken link. - I have already changed it
<img width="628" alt="Screenshot 2024-01-19 194252" src="https://github.com/sympy/sympy/assets/97588030/e041cf73-a8ff-449f-945b-6b073af4af9d">

- Why did you fix "http://www.eecis.udel.edu/~saunders/papers/sffge/it5.ps" ? - This link is downloading a file which is insecure

<img width="587" alt="Screenshot 2024-01-19 195157" src="https://github.com/sympy/sympy/assets/97588030/47bb422c-2639-4113-aa54-eced875572c2">

so, I decided to change it, but I replaced with with a wrong link. I have changed it with the new one.

<img width="657" alt="Screenshot 2024-01-19 195456" src="https://github.com/sympy/sympy/assets/97588030/005c65e1-f8f3-464f-a23f-4a44f0ff26f6">

this works fine

<img width="580" alt="Screenshot 2024-01-19 195811" src="https://github.com/sympy/sympy/assets/97588030/e7692c90-e4c8-42d2-8b50-4bd045a44df6">

sachinSingh16-09 committed:
Update control_plots.py

haru-44 commented on doc/src/contributing/new-contributors-guide/dev-setup.md:
```diff
@@ -106,19 +106,19 @@ to do them again.
 To establish a secure connection between your computer and GitHub see detailed
-instructions in <https://help.github.com/en/articles/set-up-git> or at
-<https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account>.
+instructions in <https://docs.github.com/get-started/quickstart/set-up-git> or at
```
```suggestion
instructions in <https://docs.github.com/get-started/getting-started-with-git/set-up-git> or at
```

haru-44 commented on doc/src/modules/physics/mechanics/autolev_parser.rst:
```diff
@@ -564,7 +564,7 @@ Future Improvements
 -----------------------------
 The parser has been built by referring to and parsing codes from the
-`Autolev Tutorial <https://mae.ufl.edu/~fregly/PDFs/autolev_tutorial.pdf>`_
+`Autolev Tutorial <https://mae.ufl.edu/>`_
```
Not a proper link fix.

haru-44 replied:
```suggestion
`Autolev Tutorial <https://mae.ufl.edu/~fregly/PDFs/autolev_tutorial.pdf>`_
```

If you are unsure of the appropriate link, you may leave it alone. Let's merge other fixes.

haru-44 commented on sympy/physics/control/control_plots.py:
```diff
@@ -483,7 +483,7 @@ def impulse_response_plot(system, color='b', prec=8, lower_limit=0,
     ==========
 
-    .. [1] https://www.mathworks.com/help/control/ref/lti.impulse.html
+    .. [1] https://www.mathworks.com/help/control/ref/dynamicsystem.impulse.html;jsessionid=30cd69fd0823a5e36e760497631b
```
```suggestion
    .. [1] https://www.mathworks.com/help/control/ref/dynamicsystem.impulse.html
```

sachinSingh16-09 committed:
Update doc/src/contributing/new-contributors-guide/dev-setup.md

Co-authored-by: haru-44 <36563693+haru-44@users.noreply.github.com>

sachinSingh16-09 committed:
Update sympy/physics/control/control_plots.py

Co-authored-by: haru-44 <36563693+haru-44@users.noreply.github.com>

haru-44 committed:
Update doc/src/modules/physics/mechanics/autolev_parser.rst
